cns/NetworkContainerContract.go

```diff
@@ -79,22 +80,28 @@ const (
 
 // CreateNetworkContainerRequest specifies request to create a network container or network isolation boundary.
 type CreateNetworkContainerRequest struct {
-	HostPrimaryIP              string
-	Version                    string
-	NetworkContainerType       string
-	NetworkContainerid         string // Mandatory input.
-	PrimaryInterfaceIdentifier string // Primary CA.
-	AuthorizationToken         string
-	LocalIPConfiguration       IPConfiguration
-	OrchestratorContext        json.RawMessage
-	IPConfiguration            IPConfiguration
-	SecondaryIPConfigs         map[string]SecondaryIPConfig // uuid is key
-	MultiTenancyInfo           MultiTenancyInfo
-	CnetAddressSpace           []IPSubnet // To setup SNAT (should include service endpoint vips).
-	Routes                     []Route
-	AllowHostToNCCommunication bool
-	AllowNCToHostCommunication bool
-	EndpointPolicies           []NetworkContainerRequestPolicies
+	HostPrimaryIP               string
+	Version                     string
+	NetworkContainerType        string
+	NetworkContainerid          string // Mandatory input.
+	PrimaryInterfaceIdentifier  string // Primary CA.
+	AuthorizationToken          string
+	LocalIPConfiguration        IPConfiguration
+	OrchestratorContext         json.RawMessage
+	IPConfiguration             IPConfiguration
+	SecondaryIPConfigs          map[string]SecondaryIPConfig // uuid is key
+	MultiTenancyInfo            MultiTenancyInfo
+	CnetAddressSpace            []IPSubnet // To setup SNAT (should include service endpoint vips).
+	Routes                      []Route
+	AllowHostToNCCommunication  bool
+	AllowNCToHostCommunication  bool
+	EndpointPolicies            []NetworkContainerRequestPolicies
+	BackendNetworkInterfaceInfo backendNetworkInterfaceInfo
+}
+
+type backendNetworkInterfaceInfo struct {
+	BackendNetworkInterfaceID string
+	MACAddress                string
```

Comment:
MacAddress will be coming to main NetworkContainerRequest as well, can we reuse that? @rbtr is following up on CNS Swift 2.0 changes

Reply:
yes will look into that how it is used to relay to CNI while programming the NIC, is there a way to add NIC-ID anywhere?

Reply:
@rbtr  do we have the cns changes for networkcontainerrequest merged yet? thinking which fields can we re-use to represent the same fields as above

Reply:
https://github.com/Azure/azure-container-networking/pull/2250